pkg/plugins/manager/signature/manifestverifier/verifier.go

```diff
@@ -30,17 +34,70 @@ type ManifestVerifier struct {
 	cfg  *config.Cfg
 	mlog log.Logger
 
-	lock       sync.Mutex
-	cli        http.Client
-	publicKeys map[string]ManifestKeys
+	lock sync.Mutex
+	cli  http.Client
+	kv   plugins.KeyStore
+}
+
+func New(cfg *config.Cfg, mlog log.Logger, kv plugins.KeyStore) *ManifestVerifier {
+	pmv := &ManifestVerifier{
+		cfg:  cfg,
+		mlog: mlog,
+		cli:  makeHttpClient(),
+		kv:   kv,
+	}
+	if pmv.featureEnabled() {
+		go func() {
+			pmv.updateKeysPeriodically()
+		}()
```

Comment:
We have a pattern for background services in Grafana :smile:  . Can we use that one here instead?

```go
func (pmv *ManifestVerifier) Run(ctx context.Context) error {
	if !pmv.featureEnabled() {
		<-ctx.Done()
		return ctx.Err()
	}

	// initial run and logic to determine first tick interval, etc like in updateKeys()

	downloadKeysTicker := time.NewTicker(nextSendInterval)
	defer downloadKeysTicker.Stop()

	select {
		case <-downloadKeysTicker.C:
			// download keys
		case <-ctx.Done():
			// ctx done, perform any cleanup here
			return ctx.Err()
	}
	return ctx.Err()
}
```

Then register the background service in `pkg/server/backgroundsvcs/background_services.go` in `ProvideBackgroundServiceRegistry` and re-run wire

More info: https://github.com/grafana/grafana/blob/main/contribute/backend/services.md#background-services